plugins/flatpak/gs-plugin-flatpak.c

```diff
@@ -759,6 +881,21 @@ gs_plugin_update (GsPlugin *plugin,
 	if (flatpak == NULL)
 		return TRUE;
 
+	/* Is there enough disk space to download? */
+	is_auto_update = !gs_plugin_has_flags (plugin, GS_PLUGIN_FLAGS_INTERACTIVE);
+	if (!gs_flatpak_has_space_to_update (flatpak, list_tmp, is_auto_update)) {
+		g_debug ("Skipping %s: not enough space on disk",
+			 (is_auto_update ? "automatic update" : "update"));
+		if (is_auto_update) {
+			/* Do not return an error because we don't want to see a banner
+			 * showing up out of the blue for possibly multiple update failures */
+			return TRUE;
+		}
```

Comment:
I think we need a `gs_app_set_state_recover (app);` here too?

Reply:
No, because there hasn’t been a `gs_app_set_state()` call to change the state to a transient state by this point in the code.